embedded-graphics/src/primitives/arc.rs

```diff
@@ -0,0 +1,634 @@
+                return Some(point);
+            }
+        }
+
+        None
+    }
+}
+
+/// Pixel iterator for each pixel in the arc border
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    iter: DistanceIterator,
+
+    outer_threshold: u32,
+    inner_threshold: u32,
+
+    color: Option<C>,
+}
+
+impl<C> StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    fn new(styled: &Styled<Arc, PrimitiveStyle<C>>) -> Self {
+        let Styled { primitive, style } = styled;
+
+        let stroke_area = primitive.expand(style.outside_stroke_width());
+        let fill_area = primitive.shrink(style.inside_stroke_width());
+
+        let inner_threshold = circle::diameter_to_threshold(fill_area.diameter);
+        let outer_threshold = circle::diameter_to_threshold(stroke_area.diameter);
+
+        let iter = if !styled.style.is_transparent() {
+            DistanceIterator::new(&stroke_area)
+        } else {
+            DistanceIterator::empty()
+        };
+
+        Self {
+            iter,
+            outer_threshold,
+            inner_threshold,
+            color: styled.style.stroke_color,
+        }
+    }
+}
+
+impl<C> Iterator for StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        for (point, distance) in &mut self.iter {
+            if distance < self.outer_threshold && distance >= self.inner_threshold {
+                if let Some(color) = self.color {
+                    return Some(Pixel(point, color));
+                }
+            }
+        }
+
+        None
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Arc, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_arc(self)
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Arc, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledArcIterator<C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledArcIterator::new(self)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSector {
+    line_a: LinearEquation,
+    line_b: LinearEquation,
+    draw_above_a: bool,
+    draw_above_b: bool,
+    sweep: Angle,
+}
+
+impl PlaneSector {
+    fn new(arc: &Arc) -> Self {
+        let angle_start = arc.angle_start.normalize_from(-90.0.deg());
+        let angle_end = arc.angle_end().normalize_from(-90.0.deg());
+        let negative_sweep = arc.angle_sweep < Angle::zero();
+
+        Self {
+            line_a: LinearEquation::from_point_angle(arc.center(), arc.angle_start),
+            line_b: LinearEquation::from_point_angle(arc.center(), arc.angle_end()),
+            draw_above_a: (angle_start < 90.0.deg()) ^ negative_sweep,
+            draw_above_b: (angle_end >= 90.0.deg()) ^ negative_sweep,
```

Comment:
Adding constants for `90.0.deg()` (and maybe other important angles) could reduce the number of floating point operations.

Reply:
Yep, I kind of thought about adding some consts at some point without actually doing it...
Done now, for 90deg, 180deg and 360deg.
No more use of float.deg() conversion in the code base (except in tests and examples).